algorithms/linfa-clustering/src/dbscan/algorithm.rs

```diff
@@ -71,27 +74,60 @@ use linfa::{DatasetBase, Float};
 pub struct Dbscan;
 
 impl Dbscan {
-    pub fn params<F: Float>(min_points: usize) -> DbscanHyperParamsBuilder<F> {
-        DbscanHyperParams::new(min_points)
+    /// Configures the hyperparameters with the minimum number of points required to form a cluster
+    ///
+    /// Defaults are provided if the optional parameters are not specified:
+    /// * `tolerance = 1e-4`
+    /// * `dist_fn = L2Dist` (Euclidean distance)
+    /// * `nn_impl = KdTree`
+    pub fn params<F: Float>(
+        min_points: usize,
+    ) -> DbscanHyperParamsBuilder<F, L2Dist, CommonNearestNeighbour> {
+        Self::params_with_custom_fields(min_points, L2Dist, CommonNearestNeighbour::KdTree)
+    }
+
+    /// Configures the hyperparameters with the minimum number of points, a custom distance metric,
+    /// and a custom nearest neighbour algorithm
+    pub fn params_with_custom_fields<F: Float, D: Distance<F>, N: NearestNeighbour>(
+        min_points: usize,
+        dist_fn: D,
+        nn_impl: N,
+    ) -> DbscanHyperParamsBuilder<F, D, N> {
+        DbscanHyperParamsBuilder {
+            min_points,
+            tolerance: F::cast(1e-4),
+            dist_fn,
+            nn_impl,
+        }
     }
 }
 
-impl<F: Float, D: Data<Elem = F>> Transformer<&ArrayBase<D, Ix2>, Array1<Option<usize>>>
-    for DbscanHyperParams<F>
+impl<F: Float, D: Data<Elem = F>, DF: Distance<F>, N: NearestNeighbour>
+    Transformer<&ArrayBase<D, Ix2>, Array1<Option<usize>>> for DbscanHyperParams<F, DF, N>
```

Comment:
this should actually be a prediction here, because it does not transform the representation of a dataset

Reply:
What's the distinction between `predict`, `transform`, and `fit` in this context?

Reply:
* `fit` takes a training dataset and returns a model, for NN this would be all points because they are interpolated
* `predict` takes a reference to data and predict their clusters (or noise)
* `transform` takes a dataset and transform the variables into the embedding (either integers or sparse one-hot matrix)